transcom/mymove
# INT B-18475-db-backend-updates-shipment-dest-address-update-sit

INTEGRATION BRANCH

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A888121&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

In order to notify the TOO when a shipment address change request contains destination SIT items, we required additions of columns to the `shipment_address_update` table to store the data that will be needed in the UI. In addition to those database changes, backend logic was required to populate those columns with the appropriate data.

This PR does the following:
- adds the columns `sit_original_address_id`, `old_sit_distance_between`, & `new_sit_distance_between` to the `shipment_address_updates` table
- updated Go Struct for the model `ShipmentAddressUpdate`
- updated `swagger-def` for GHC API to return new data variables that will be used in the UI
- updated `mtoShipmentFetcher` to include additional data for `ShipmentAddressUpdate`
- added logic checks in `shipment_address_update_requester.go` to find, calculate, and populate data in db
- added tests to support new functionality

### How to test

1. Access MM as the prime
2. Request a shipment destination address change for a shipment that contains destination SIT
3. Load up the db and see new columns and data in `shipment_address_updates` table
